Index: src/subsets.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/subsets.java b/src/subsets.java
new file mode 100644
--- /dev/null	(date 1686764881760)
+++ b/src/subsets.java	(date 1686764881760)
@@ -0,0 +1,34 @@
+
+
+import java.util.Scanner;
+
+public class subsets {
+    public static void main(String args[]) {
+
+        Scanner sc= new Scanner(System.in);
+        int n= sc.nextInt();
+        int[] arr= new int[n];
+        for(int i=0;i<n;i++){
+            arr[i]=sc.nextInt();
+        }
+
+        int k=sc.nextInt();
+
+        subset(arr,0,k,0,"");
+
+    }
+
+    public static void subset(int[] arr,int ind,int tar,int sum,String ans){
+        if(ind==arr.length){
+            if(sum==tar){
+                System.out.println(ans);
+            }
+            return;
+        }
+        subset(arr,ind+1,tar,sum,ans);
+        subset(arr,ind+1,tar,sum+arr[ind],arr[ind]+" "+ans);
+
+    }
+
+}
+
Index: src/SplitArray.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/SplitArray.java b/src/SplitArray.java
new file mode 100644
--- /dev/null	(date 1686764964818)
+++ b/src/SplitArray.java	(date 1686764964818)
@@ -0,0 +1,35 @@
+
+
+import java.util.Scanner;
+
+public class SplitArray {
+    static int count = 0;
+
+    public static void main(String[] args) {
+        count = 0;
+        Scanner sc = new Scanner(System.in);
+
+        int n = sc.nextInt();
+        int[] arr = new int[n];
+
+        for (int i = 0; i < n; i++) {
+            arr[i] = sc.nextInt();
+        }
+
+        disp(0, "", 0, "", 0, arr);
+        System.out.println(count);
+    }
+
+    public static void disp(int i, String str1, int sum1, String str2, int sum2, int[] arr) {
+        if (i == arr.length) {
+            if (sum1 == sum2) {
+                System.out.println(str1 + "and " + str2);
+                count++;
+            }
+            return;
+        }
+        disp(i+1, str1 + arr[i] + " ", sum1 + arr[i], str2, sum2, arr);
+        disp(i+1, str1, sum1, str2+arr[i] + " ", sum2+arr[i],arr);
+}
+}
+
